rtbkit/common/filter.h

```diff
@@ -424,6 +424,17 @@ struct FilterState
     // creative matrix. This is the format ingested by the router.
     std::unordered_map<unsigned, BiddableSpots> biddableSpots();
 
+    /*
+     * This map is keyed by filtered reasons and contains a list of ConfigIndex.
+     * Of course, those configs belonging to those configIndex were filtered
+     * due to the given reason.
+     */
+    typedef std::map<std::string, std::list<unsigned> > FilterReasons;
```

Comment:
I would use a ConfigSet instead of an std::list here because otherwise, any filter that wants to construct a reason, will have to do the conversion from ConfigSet as opposed to just having the FilterPool do the conversion for everybody. 

In the case of iterative filters, we could always modify the interface to return a string as well as the bool which would be used by the base filter to fill in the reason if a string is returned. Note that we'll make this change as needed.